src/Avalonia.Diagnostics/Diagnostics/Views/MainWindow.xaml.cs

```diff
@@ -161,8 +161,19 @@ private void RawKeyDown(RawKeyEventArgs e)
                 return;
             }
 
-            var root = Root as TopLevel
-                ?? vm.PointerOverRoot as TopLevel;
+            var root = vm.PointerOverRoot as TopLevel;
+            IControl? parent = root;
+
+            while (parent?.Parent != null)
+            {
+                parent = parent.Parent;
+            }
```

Comment:
Why not to use VisualRoot property on control?

Reply:
Not sure why, but `VisualRoot `on the `PopupRoot` was always null. I guess I could somehow access the `_parent` field of `PopupRoot`?

Reply:
I exposed the field now. That seems to be the cleanest solution here IMO.